core/src/replay_stage.rs

```diff
@@ -306,7 +306,7 @@ impl ReplayStage {
                     // Vote on a fork
                     let voted_on_different_fork = {
                         if let Some(ref vote_bank) = vote_bank {
-                            subscriptions.notify_subscribers(vote_bank.slot(), &bank_forks);
+                            subscriptions.notify_subscribers(block_commitment_cache.read().unwrap().slot(), &bank_forks);
```

Comment:
I think the block commitment cache will always be a bit behind. Seems like subscribers won't be notified about the latest confirmation counts until the next vote.

What if we hooked up a channel between the subscriptions loop and the `AggregateCommitmentService` and passed the whole `Arc<RwLock<BlockCommitmentCache>>` whenever it's updated?

Reply:
Yes, I think it makes sense to move to something like that. As discussed in PM, will implement in a follow-up pr!

Reply:
Yeah it's true the commitment cache is only updated when this validator votes. This might be a problem if other validators in the network are updating, but this validator isn't voting (maybe some sort of partition), should be rare though...